### PR TITLE
update simplecov to latest version

### PR DESCRIPTION
**Pull request by rileyanderson**
## Description of change
As part of the gem update series, this PR updates the simplecov gem to the latest version. Need to insure that CodeClimate still works after updating this.
Updates:
`simplecov -> 0.21.2`
`docile -> 1.3.5`
`json- > 2.5.1`
`simplecov-html -> 0.12.3`
`simplecov_json_formatter -> 0.1.2`

## Original issue(s)
department-of-veterans-affairs/va.gov-team#20261

### CODE REVIEW

**Comment by LindseySaari** [comment]
Note: I would also check to make sure that this doesn't break anything in the CircleCI run as well

**Comment by rileyanderson** [comment]
@LindseySaari confirmed this works on circle still. This is ready for review

File: spec/simplecov_helper.rb [hunk 1]
```diff
@@ -32,10 +32,10 @@ def all_results
   end
 
   def merge_results
-    results = all_results.map { |file| SimpleCov::Result.from_hash(JSON.parse(File.read(file))) }
-    SimpleCov::ResultMerger.merge_results(*results).tap do |result|
-      SimpleCov::ResultMerger.store_result(result)
-    end
+    SimpleCov.collate all_results
+  rescue RuntimeError => e
+    puts e
```
Comment (LindseySaari) [inline spec/simplecov_helper.rb:37]: Looks like a debug statement 😸

File: Gemfile.lock [hunk 2]
```diff
@@ -295,7 +295,7 @@ GEM
     descendants_tracker (0.0.4)
       thread_safe (~> 0.3, >= 0.3.1)
     diff-lcs (1.4.4)
-    docile (1.3.2)
+    docile (1.3.5)
```
Comment (LindseySaari) [inline Gemfile.lock:298]: Could you also note the gem dependencies of simplecov that had to be updated in the PR description please. Looks like docile, simplecov-html and simplecov_json_formatter

File: Makefile [hunk 3]
```diff
@@ -149,7 +149,6 @@ ifeq ($(ENV_ARG), dev)
 	@$(BASH_DEV) "RAILS_ENV=test DISABLE_BOOTSNAP=true NOCOVERAGE=true bundle exec parallel_rspec ${SPEC_PATH}"
 else
 	@$(COMPOSE_TEST) $(BASH) -c "CIRCLE_JOB=true RAILS_ENV=test DISABLE_BOOTSNAP=true bundle exec parallel_rspec ${SPEC_PATH}"
-	@$(BASH_TEST) "DISABLE_BOOTSNAP=true RUN_COVERAGE=true bundle exec rake simplecov:report_coverage"
```
Comment (LindseySaari) [inline Makefile:152]: Should this be removed?

Comment (LindseySaari) [reply Makefile:152]: Oh, so it looks like if we're not using this anymore, can we just completely remove simplecov_parallel.rake and the simplecov_helper

Comment (rileyanderson) [reply Makefile:152]: From what I could tell it wasn't doing anything. That rake task is specific to circleci file path for coverage and cirlceci doesn't use the this makefile only Jenkins does. 

CircleCI reports coverage here: https://github.com/department-of-veterans-affairs/vets-api/blob/master/.circleci/config.yml#L111  

Jenkins does it here: https://github.com/department-of-veterans-affairs/vets-api/blob/master/Jenkinsfile#L81-L84

Comment (rileyanderson) [reply Makefile:152]: CircleCI still uses `simplecov_parallel.rake` and the `simplecov_helper`

Comment (LindseySaari) [reply Makefile:152]: Ah, ok! Makes sense!

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**